### DISCUSSION TITLE
Shelly i4 plus MATTER

### DISCUSSION THREAD

**Discussion by goeran03** [arendst/Tasmota#18979] [Q&A]
I don't see the switches of a shelly i4 in homekit. I use tasmota 13.0.0. There is only the bridge but with no endpoints connected. Even if I try to add local endpoint on the tasmota Matter-Configuration menu of type "OnOffSensor" I don't see any reaction. In Status 8 the switches are visible on tasmota's console. Any suggestion?
Thanks Goeran

**Comment by s-hadinger** [comment]
Hi. I don't want to seem rude, but you gave zero details about how you configured the template for your Shelly, zero detail about how you configured Matter endpoints and zero logs. We can't help you trying to guess how is your configuration.

**Comment by goeran03** [comment]
What actually are different options for this? However, here we go:
{"NAME":"Shelly Plus i4","GPIO":[0,0,0,0,0,0,0,0,192,0,193,0,0,0,0,0,0,0,0,0,0,0,195,194,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}
Can I provide anything more?

**Reply by s-hadinger** [reply]
Thanks. I don't think that Apple currently supports OnOffSwitch at all. You can set a switch as an occupancy sensor or as a contact sensor

**Reply by goeran03** [reply]
Will try this thanks.

**Reply by goeran03** [reply]
At least IOS supports "stateless switches" and "stateful switches" as well according to their developer's documentation for homekit.  An occupancy sensor seems not to be the solution.

Thanks for your work. G.

**Comment by goeran03** [comment]
By the way, the type of sensor does not seem to be the problem. Neither occupancy nor contact solve it. My experience so far was, that a temperature sensor for instance showed up in matter configuration automatically if configured in template before. Not so the switches here. Trying  to add any type of switch sensor manually also not lead to success. At least here. 
    	
    
  

 Von: s-hadinger ***@***.***>Gesendet: Dienstag, Juni 27, 2023 11:30 PMAn: arendst/Tasmota ***@***.***>Cc: goeransaenger ***@***.***>; Author ***@***.***>Betreff: Re: [arendst/Tasmota] Shelly i4 plus MATTER (Discussion #18979) 
Thanks. I don't think that Apple currently supports OnOffSwitch at all. You can set a switch as an occupancy sensor or as a contact sensor

**Comment by kibeki** [comment]
Hi, which tasmota firmware file did you use to flash a Shelly Plus i4?

**Reply by Jason2866** [reply]
tasmota32-solo1 is the one.

**Reply by kibeki** [reply]
but the shelly needs a zip-file not only the bin
do you have a tipp how to do the flashing with only have a bin-file?

**Reply by Jason2866** [reply]
https://github.com/tasmota/mgos32-to-tasmota32
